app/src/main/java/com/project/app/base/BaseFragment.kt

```diff
@@ -0,0 +1,158 @@
+package com.project.app.base
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.os.Bundle
+import android.view.*
+import android.view.inputmethod.InputMethodManager
+import android.widget.ImageButton
+import android.widget.TextView
+import android.widget.Toast
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
+import androidx.viewbinding.ViewBinding
+import com.project.app.R
+import com.project.app.helpers.LoadingDialog
+import com.project.app.helpers.ToolbarType
+import kotlinx.coroutines.Job
+
+abstract class BaseFragment<VB : ViewBinding> : Fragment(), IBaseFragment {
+    lateinit var activityIndicator: LoadingDialog
+    private var job = Job()
+    private lateinit var topBarLayout: ConstraintLayout
+    lateinit var leftButtonToolbar: ImageButton
+    private lateinit var textViewMenuTitle: TextView
+    private lateinit var rigthButtonToolbar: ImageButton
+
+    private lateinit var fragmentView: View
+
+    private var _binding: ViewBinding? = null
+    abstract val bindingInflater: (LayoutInflater, ViewGroup?, Boolean) -> VB
+    @Suppress("UNCHECKED_CAST")
+    val binding: VB
+        get() = _binding as VB
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = bindingInflater.invoke(inflater, container, false)
+        return requireNotNull(_binding).root
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        topBarLayout = requireActivity().findViewById(R.id.topBarLayout)
+        leftButtonToolbar = requireActivity().findViewById(R.id.leftButtonToolbar)
+        rigthButtonToolbar = requireActivity().findViewById(R.id.rigthButtonToolbar)
+        textViewMenuTitle = requireActivity().findViewById(R.id.menuItemTitle)
```

Comment:
En vez de hacer varios llamados a `requireActivity`, por qué no llamarlo una sola vez y guardarlo en una variable? Es decir, algo así...

```suggestion
        val activity = requireActivity()
        topBarLayout = activity.findViewById(R.id.topBarLayout)
        leftButtonToolbar = activity.findViewById(R.id.leftButtonToolbar)
        rigthButtonToolbar = activity.findViewById(R.id.rigthButtonToolbar)
        textViewMenuTitle = activity.findViewById(R.id.menuItemTitle)
```

De esta forma, nos evitamos el llamado a una misma función que puede tener su costo.

Otra forma de hacerlo, es usando los poderes de programación funcional de Kotlin. Sería algo así...

```suggestion
        requireActivity().run {
            topBarLayout = findViewById(R.id.topBarLayout)
            leftButtonToolbar = findViewById(R.id.leftButtonToolbar)
            rigthButtonToolbar = findViewById(R.id.rigthButtonToolbar)
            textViewMenuTitle = findViewById(R.id.menuItemTitle)
        }
```

Reply:
_"val activity = requireActivity()"_ sí, realmente fue otra cosa arrastrada del pasado jaja. De todas formas no conocía esa magia de kotlin, me gusta más